Source/WebKit/Shared/SessionState.h

```diff
@@ -62,15 +63,21 @@ struct HTTPBody {
     Vector<Element> elements;
 };
 
-class FrameState {
+class FrameState : public RefCounted<FrameState> {
 public:
+    static Ref<FrameState> create();
+    static Ref<FrameState> create(const String& urlString, const String& originalURLString, const String& referrer, const AtomString& target, std::optional<WebCore::FrameIdentifier>, std::optional<Vector<uint8_t>> stateObjectData, int64_t documentSequenceNumber, int64_t itemSequenceNumber, WebCore::IntPoint scrollPosition, bool shouldRestoreScrollPosition, float pageScaleFactor, std::optional<HTTPBody>&&, WebCore::BackForwardItemIdentifier, bool hasCachedPage, const String& title, WebCore::ShouldOpenExternalURLsPolicy, RefPtr<WebCore::SerializedScriptValue>&& sessionStateObject, bool wasCreatedByJSWithoutUserInteraction,
```

Comment:
This is a prime candidate for variadic templates.  CSSColor is an example of one where we already have this.

Reply:
good idea. fixed.